python/manylinux1/scripts/build_boost.sh

```diff
@@ -20,10 +20,16 @@ BOOST_VERSION=1.73.0
 BOOST_VERSION_UNDERSCORE=${BOOST_VERSION//\./_}
 NCORES=$(($(grep -c ^processor /proc/cpuinfo) + 1))
 
-curl -sL https://sourceforge.net/projects/boost/files/boost/${BOOST_VERSION}/boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 -o /boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2
-tar xf boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2
+BASE_NAME=boost_${BOOST_VERSION_UNDERSCORE}
+ARCHIVE_NAME=boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2
+
+# Bintray is much faster but can fail because of limitations
+curl -sL https://dl.bintray.com/boostorg/release/${BOOST_VERSION}/source/boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 -o /${ARCHIVE_NAME} \
+    || curl -sL https://sourceforge.net/projects/boost/files/boost/${BOOST_VERSION}/boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 -o /${ARCHIVE_NAME}
```

Comment:
Nice!

Reply:
FYI: I've created https://bintray.com/apache/arrow/boost-mirror and mirrored Boost 1.73.0 as https://apache.bintray.com/arrow/boost/1.73.0/boost_1_73_0.tar.bz2 .

We can use this to save the official Boost download resource.